src/v/pandaproxy/schema_registry/types.h

```diff
@@ -271,6 +272,11 @@ class protobuf_schema_definition {
     ::result<ss::sstring, kafka::error_code>
     name(const std::vector<int>& fields) const;
 
+    ::result<
+      std::reference_wrapper<const google::protobuf::Descriptor>,
+      kafka::error_code>
+    descriptor(const std::vector<int>& fields) const;
```

Comment:
I think it would be cleaner to move this to `protobuf.h`.
```c++
::result<
  std::reference_wrapper<const google::protobuf::Descriptor>,
  kafka::error_code>
descriptor(const protobuf_schema_definition&, const std::vector<int>& fields);
```

Reply:
Sure, will need to take the pimpl though unless you would prefer I use friendship

Reply:
Ah thanks, will update

Reply:
Done